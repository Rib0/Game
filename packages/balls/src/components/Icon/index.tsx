import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export interface IIconProps {
    icon: 'sync-alt' | 'reply' | 'plus' | 'palette';
}

export const Icon = ({ icon }: IIconProps) => <FontAwesomeIcon icon={icon} />;
