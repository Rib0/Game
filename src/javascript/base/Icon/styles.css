.icon {
    cursor: pointer;
    color: var(--colorWhite);
    transition: transform 0.2s linear;
}
