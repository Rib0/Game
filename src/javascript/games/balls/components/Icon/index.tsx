import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export interface IIconProps {
    icon: 'sync-alt' | 'reply' | 'plus';
};

const Icon = ({ icon }: IIconProps) => {
    return <FontAwesomeIcon icon={icon} />;
};

export default Icon;
