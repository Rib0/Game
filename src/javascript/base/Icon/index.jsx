import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from './styles.css';

const cx = classNames.bind(styles);

class Icon extends PureComponent {
    static IconTypes = {
        faBars: 'bars',
        faHome: 'home',
    };

    static IconSizes = {
        large: '3x',
    };

    render() {
        const { type, size, className } = this.props;

        return (
            <div className={cx('icon', className)}>
                <FontAwesomeIcon size={size} icon={type} />
            </div>
        );
    }
}

Icon.propTypes = {
    type: PropTypes.oneOf(Object.values(Icon.IconTypes)),
    size: PropTypes.oneOf(Object.values(Icon.IconSizes)),
    iconClassName: PropTypes.string,
};

Icon.defaultProps = {
    size: Icon.IconSizes.large,
};

export default Icon;
