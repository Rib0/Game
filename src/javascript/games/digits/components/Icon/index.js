import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './styles.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'; // todo сделать Button component with Icon, убрать margin у button

class Icon extends PureComponent {

    static IconTypes = {
        faBars: 'bars'
    }

    // todo добавить цвета

    static IconSizes = {
        large: '3x'
    }

    render() {
        const { type, size, onClick, rotated } = this.props;

        return (
            <div onClick={onClick} className={cx(styles.icon, rotated && styles.rotated)}>
                <FontAwesomeIcon size={size} icon={type} />
            </div>
        )
    }
}

Icon.propTypes = {
    type: PropTypes.oneOf(Object.values(Icon.IconTypes)),
    size: PropTypes.oneOf(Object.values(Icon.IconSizes)),
    onClick: PropTypes.func
};

export default Icon;