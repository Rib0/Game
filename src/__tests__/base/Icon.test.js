import React from 'react';
import Icon from 'base/Icon';

const setUp = props => shallow(<Icon {...props} />);

describe('Icon test', () => {
    const props = {
        type: Icon.IconTypes.faBars,
        size: Icon.IconSizes.large,
        className: 'customClassName',
    };

    test('Icon should render', () => {
        const Component = setUp(props);
        expect(Component).toMatchSnapshot();
    });
    test('Icon should throw type prop', () => {
        const Component = setUp(props);
        expect(Component.find('FontAwesomeIcon').prop('icon')).toBe(Icon.IconTypes.faBars);
    });
    test('Icon should throw size prop', () => {
        const Component = setUp(props);
        expect(Component.find('FontAwesomeIcon').prop('size')).toBe(Icon.IconSizes.large);
    });
});
