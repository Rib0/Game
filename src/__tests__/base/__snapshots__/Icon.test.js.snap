// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test Icon should render 1`] = `
<div
  className="icon customClassName"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="bars"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="3x"
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;
